perllib/FixMyStreet/SMS.pm

```diff
@@ -54,21 +73,36 @@ sub send {
     my ($self, %params) = @_;
 
     my $twilio = $self->twilio;
-    unless ($twilio) {
+    my $notify = $self->notify;
+    unless ($twilio || $notify) {
         return { error => "No SMS service configured" };
     }
 
-    my $output = $twilio->{api}->POST('Messages.json',
-        $twilio->{from} ? (From => $twilio->{from}) : (),
-        $twilio->{messaging_service} ? (MessagingServiceSid => $twilio->{messaging_service}) : (),
-        To => $params{to},
-        Body => $params{body},
-    );
-    my $data = decode_json($output->{content});
-    if ($output->{code} >= 400) {
-        return { error => "$data->{message} ($data->{code})" };
+    if ($twilio) {
+        my $output = $twilio->{api}->POST('Messages.json',
+            $twilio->{from} ? (From => $twilio->{from}) : (),
+            $twilio->{messaging_service} ? (MessagingServiceSid => $twilio->{messaging_service}) : (),
+            To => $params{to},
+            Body => $params{body},
+        );
+        my $data = decode_json($output->{content});
+        if ($output->{code} >= 400) {
+            return { error => "$data->{message} ($data->{code})" };
+        }
+        return { success => $data->{sid} };
+    }
+
+    if ($notify) {
+        my $output = $notify->send(
+            to => $params{to},
+            body => $params{body},
+        );
+        my $data = decode_json($output->{content});
+        if ($output->{code} >= 400) {
+            return { error => "$data->[0]{message} ($data->[0]{error})" };
+        }
+        return { success => $data->{id} };
```

Comment:
Although, thinking about this, if we ever did turn on Twilio for FixMyStreet then I think this would mean we'd send out messages for councils with gov.notify twice?

Reply:
No, if twilio is true the above if statement always returns there.
I should swap them, because yes, I doubt you'd ever have both but if you did and a cobrand had Notify you'd presumably want to use that.

Reply:
oh yes, return. doh!